src/core/meshcop/commissioner.cpp

```diff
@@ -81,6 +84,29 @@ Commissioner::Commissioner(Instance &aInstance)
     mCommissionerAloc.mScopeOverrideValid = true;
 }
 
+void Commissioner::SetState(otCommissionerState aState)
+{
+    VerifyOrExit(mState != aState);
+
+    mState = aState;
+
+    if (mStateCallback)
+    {
+        mStateCallback(mState, mCallbackContext);
+    }
+
```

Comment:
How about moving the `Get<Notifier>().Signal(OT_CHANGED_COMMISSIONER_STATE);` from different methods here?

Reply:
I was hoping to remove the `OT_CHANGED_COMMISSIONER_STATE` (and `OT_CHANGED_JOINER_STATE`).

Reply:
I updated this PR to remove `OT_CHANGED_COMMISSIONER_STATE` and `OT_CHANGED_JOINER_STATE`.